src/node/ui_interface.h

```diff
@@ -25,8 +25,7 @@ class CClientUIInterface
 {
 public:
     /** Flags for CClientUIInterface::ThreadSafeMessageBox */
-    enum MessageBoxFlags
-    {
+    enum MessageBoxFlags : unsigned {
```

Comment:
`unsigned` is technically only guaranteed to be 16-bit, but `StandardButton` has values approaching 32-bit, so IMO better to specify `unsigned long` or `uint32_t` here.

Reply:
Bitcoin Core can't start nor compile with 16-bit int. See also: 

```
src/compat/assumptions.h:static_assert(sizeof(short) == 2, "16-bit short assumed");
src/compat/assumptions.h:static_assert(sizeof(int) == 4, "32-bit int assumed");
src/compat/assumptions.h:static_assert(sizeof(unsigned) == 4, "32-bit unsigned assumed");
src/compat/assumptions.h:static_assert(sizeof(size_t) == 4 || sizeof(size_t) == 8, "size_t assumed to be 32-bit or 64-bit");

Reply:
Yes, but that's not a reason to make it worse when it could easily be made well-defined.

Reply:
Done, switched to `uint32_t`.